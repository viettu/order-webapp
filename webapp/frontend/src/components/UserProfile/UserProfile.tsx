import {
  Avatar,
  Box,
  Flex,
  HStack,
  VStack,
  Text,
  Menu,
  MenuButton,
  MenuDivider,
  MenuItem,
  MenuList,
} from '@chakra-ui/react';
import { FiChevronDown } from 'react-icons/fi';
import { Link } from 'react-router-dom';

export default function UserProfile() {
  return (
    <HStack spacing={{ base: '0', md: '6' }}>
      <Flex alignItems="center">
        <Menu>
          <MenuButton py={2} transition="all 0.3s" _focus={{ boxShadow: 'none' }}>
            <HStack spacing="4">
              <Avatar size="md" name="Eric Nguyen" />
              <VStack display={{ base: 'none', md: 'flex' }} alignItems="flex-start" spacing="1px" ml="2">
                <Text fontSize="md" color="gray.600">
                  Eric Nguyen
                </Text>
              </VStack>
              <Box display={{ base: 'none', md: 'flex' }}>
                <FiChevronDown />
              </Box>
            </HStack>
          </MenuButton>
          <MenuList fontSize="lg" bg="white" borderColor="gray.200">
            <MenuItem>Profile</MenuItem>
            <MenuDivider />
            <MenuItem>
              <Link to={'/orders'}>Your orders</Link>
            </MenuItem>
            <MenuDivider />
            <MenuItem>Sign out</MenuItem>
          </MenuList>
        </Menu>
      </Flex>
    </HStack>
  );
}
