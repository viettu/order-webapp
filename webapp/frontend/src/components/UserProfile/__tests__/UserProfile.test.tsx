import React from 'react';
import { UserProfile } from '../UserProfile';
import renderer from 'react-test-renderer';
import { TestComponentWrapper } from '../../../utils';

test('snapshot', () => {
  const component = renderer.create(
    <TestComponentWrapper>
      <UserProfile></UserProfile>
    </TestComponentWrapper>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
