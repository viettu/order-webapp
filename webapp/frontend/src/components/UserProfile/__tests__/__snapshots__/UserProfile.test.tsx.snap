// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <div
    className="chakra-stack css-1ggbuc9"
  >
    <div
      className="css-70qvj9"
    >
      <button
        aria-controls="menu-list-:r1:"
        aria-expanded={false}
        aria-haspopup="menu"
        className="chakra-menu__menu-button css-vn8yib"
        id="menu-button-:r1:"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <span
          className="css-xl71ch"
        >
          <div
            className="chakra-stack css-a9v878"
          >
            <span
              className="chakra-avatar css-wrgj4r"
            >
              <div
                aria-label="Eric Nguyen"
                className="chakra-avatar__initials css-uo84k3"
                role="img"
              >
                EN
              </div>
            </span>
            <div
              className="chakra-stack css-1meoifv"
            >
              <p
                className="chakra-text css-rukqh5"
              >
                Eric Nguyen
              </p>
            </div>
            <div
              className="css-dho3mt"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="6 9 12 15 18 9"
                />
              </svg>
            </div>
          </div>
        </span>
      </button>
      <div
        className="css-r6z5ec"
        style={
          Object {
            "inset": "0 auto auto 0",
            "minWidth": "max-content",
            "position": "absolute",
            "visibility": "hidden",
          }
        }
      >
        <div
          aria-orientation="vertical"
          className="chakra-menu__menu-list css-dthcfm"
          id="menu-list-:r1:"
          onKeyDown={[Function]}
          role="menu"
          style={
            Object {
              "opacity": 0,
              "transform": "scale(0.8) translateZ(0)",
              "transformOrigin": "var(--popper-transform-origin)",
              "visibility": "hidden",
            }
          }
          tabIndex={-1}
        >
          <hr
            aria-orientation="horizontal"
            className="chakra-menu__divider css-vfkdr8"
            role="separator"
          />
          <a
            href="/orders"
            onClick={[Function]}
          >
            <button
              className="chakra-menu__menuitem css-13c7rae"
              id="menu-list-:r1:-menuitem-:r2:"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              role="menuitem"
              tabIndex={0}
              type="button"
            >
              Your orders
            </button>
          </a>
          <hr
            aria-orientation="horizontal"
            className="chakra-menu__divider css-vfkdr8"
            role="separator"
          />
          <button
            className="chakra-menu__menuitem css-13c7rae"
            id="menu-list-:r1:-menuitem-:r3:"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            Sign out
          </button>
        </div>
      </div>
    </div>
  </div>,
  <span />,
  <span
    className="chakra-env"
    hidden={true}
  />,
]
`;
